File: Ruby/Chapter02/knock13.rb
# impliment unix command 'paste'
# this command is connecting each row


col1, col2 = open("../data/col1.txt"), open("../data/col2.txt")
merged = open("../data/merged.tsv", 'w')
while true
  line1, line2 = col1.gets, col2.gets
  if line1 == nil
    break
  end
  merged.write(line1.strip() + "\t" + line2)
end

File: Csharp/chapter02/knock13.cs
/*
 * Chapter02, knock13
 * Task: merge two column
 *       Unix Command ``paste''
 */

using System;
using System.IO;
using System.Collections.Generic;

public class Knock13 {
    static public void Main(string[] args) {
        MeargeColumn(args);
    }

    public static void MeargeColumn(string[] paths) {
        using (StreamReader sr1 = File.OpenText(paths[0]), sr2 = File.OpenText(paths[1])) {
            string line1 = "", line2 = "";
            using (StreamWriter sw = File.CreateText("./data/col12.txt")) {
                while ((line1 = sr1.ReadLine()) != null && (line2 = sr2.ReadLine()) != null) {
                    sw.WriteLine(string.Format("{0}\t{1}", line1, line2));
                }   
            }
        }
    }
    
}

